mtags/src/main/scala/scala/meta/internal/pc/Completions.scala

```diff
@@ -669,6 +676,42 @@ trait Completions { this: MetalsGlobal =>
     }
   }
 
+  /**
+   * Find a method definition right after the given position.
+   *
+   * @param pos The position of scaladoc in the original source.
+   *            This class will find the associated member def based on this pos.
+   */
+  class AssociatedMemberDefFinder(pos: Position) extends Traverser {
+    private var defs: List[MemberDef] = Nil
+
+    /**
+     * Collect all the member definitions whose position is
+     * below the given `pos`. And then return the closest member definiton.
+     */
+    def findAssociatedDef(root: Tree): Option[MemberDef] = {
+      defs = Nil
+      traverse(root)
+      defs.sortBy(_.pos.point).headOption
+    }
+    override def traverse(t: Tree): Unit = {
```

Comment:
Since just traversing the AST for the source, if the source has long lines, it takes a bit time before returning the completion response...